test/unit/edition_test.rb

```diff
@@ -904,4 +904,22 @@ class EditionTest < ActiveSupport::TestCase
     refute edition.historic?
   end
 
+  test '#has_document? returns true if has document' do
+    edition = build(:edition)
+    refute edition.has_document?
+
+    edition = create(:edition)
```

Comment:
Looks a bit implicit here that saving creates a document, I'll update this I think.